presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,355 @@ connection properties as appropriate for your setup:
+To see a list of the columns in the ``clicks`` table in the ``web``
+database, run either of the following::
 
     DESCRIBE oracle.web.clicks;
     SHOW COLUMNS FROM oracle.web.clicks;
 
-Finally, you can access the ``clicks`` table in the ``web`` database::
+To access the clicks table in the web database, run the following::
 
     SELECT * FROM oracle.web.clicks;
 
-If you used a different name for your catalog properties file, use
-that catalog name instead of ``oracle`` in the above examples.
+Mapping data types between Presto and Oracle
+--------------------------------------------
+
+Both Oracle and Presto have types that are not supported by the Oracle
+connector. The following sections explain their type mapping.
+
+Oracle to Presto type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports selecting Oracle database types. This table shows the Oracle to
+Presto data type mapping:
+
+
+.. list-table:: Oracle to Presto type mapping
+  :widths: 20, 20, 60
+  :header-rows: 1
+
+  * - Oracle database type
+    - Presto type
+    - Notes
+  * - ``NUMBER(p, s)``
+    - ``DECIMAL(p, s)``
+    -  See :ref:`number mapping`
+  * - ``NUMBER(p)``
+    - ``DECIMAL(p, 0)``
+    - See :ref:`number mapping`
+  * - ``FLOAT[(p)]``
+    - ``DOUBLE``
+    -
+  * - ``BINARY_FLOAT``
+    - ``REAL``
+    -
+  * - ``BINARY_DOUBLE``
+    - ``DOUBLE``
+    -
+  * - ``VARCHAR2(n CHAR)``
+    - ``VARCHAR(n)``
+    -
+  * - ``VARCHAR2(n BYTE)``
+    - ``VARCHAR(n)``
+    -
+  * - ``NVARCHAR2(n)``
+    - ``VARCHAR(n)``
+    -
+  * - ``CHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``NCHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``CLOB``
+    - ``VARCHAR``
+    -
+  * - ``NCLOB``
+    - ``VARCHAR``
+    -
+  * - ``RAW(n)``
+    - ``VARBINARY``
+    -
+  * - ``BLOB``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p)``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p) WITH TIME ZONE``
+    - ``TIMESTAMP WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+If an Oracle table uses a type not listed in the above table, then you can use the
+``unsupported-type.handling`` configuration property to specify Presto behavior.
+For example:
+
+- If ``unsupported-type.handling`` is set to ``FAIL``, then the
+  querying of an unsupported table fails.
+- If ``unsupported-type.handling`` is set to ``IGNORE``, 
+  then you can't see the unsupported types in Presto.
+- If ``unsupported-type.handling`` is set to ``CONVERT_TO_VARCHAR``, 
+  then the column is exposed as unbounded ``VARCHAR``.
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE <table> AS SELECT`` operations. When data is inserted into existing
```

Comment:
This sentence was suggested by @losipiuk .. however I am not sure this actually true. At least all other connector docs suggest that the Presto to Oracle mapping is used for ALL write operations .. and not just for CTAS ... 

Can you clarify if the Oracle connector is just different or if all other docs are wrong or whatever is technically correct @losipiuk @findepi and @electrum

Reply:
The "Presto to <some DB>" mapping is used only in CTAS flow, when the external table does not yet exist, and we need to build the a schema for it based on Presto types of data used in CTAS statement.
For query operations as well as INSERT operations the "<some DB> to Presto" mapping is used. We already have columns in external database and we map types of those columns to Presto types.
This is not Oracle specific, but true for all connectors based on presto-base-jdbc.

Still I am not at all defending the wording I suggested. It is highly probable that one can phrase it much better :)

Hera are the code pointers:
* CTAS: https://github.com/prestosql/presto/blob/fd66bb72ddb9cfc311511900b1454cef385f172d/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java#L74
* INSERT: https://github.com/prestosql/presto/blob/fd66bb72ddb9cfc311511900b1454cef385f172d/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java#L89

Reply:
So looking at the code I conclude that this is true for all connectors that use the base-jdbc stuff .. so including mysql, postgresql, oracle, sqlserver and maybe others. 

At this stage I suggest we leaven the wording as it is in this PR and create another PR that clarifies this and corrects the docs for all affected connectors. Do you agree with this approach @losipiuk  and @electrum  .. if yes .. can we get this PR merged.

Reply:
SGTM